websites/H/HentaiZ/presence.ts

```diff
@@ -0,0 +1,97 @@
+const presence = new Presence({
+    clientId: "826391938542207018"
+  }),
+  strings = presence.getStrings({
+    play: "presence.playback.playing",
+    pause: "presence.playback.paused"
+  });
+
+let lastPlaybackState: boolean,
+  lastPath: string, video = {
+    duration: 0,
+    currentTime: 0,
+    paused: true
+  },
+  browsingStamp = Math.floor(Date.now() / 1000);
+
+presence.on(
+  "iFrameData",
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
+    video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
+  const curPath = document.location.pathname,
+    playback = video !== null && !isNaN(video.duration) && curPath.includes("/xem-phim"),
+    presenceData: PresenceData = {
+      largeImageKey: "logo"
+    }, search = document.location.search;
+
+  if (lastPath !== curPath || lastPlaybackState !== playback) {
+    lastPath = curPath;
+    lastPlaybackState = playback;
+    browsingStamp = Math.floor(Date.now() / 1000);
+  }
+
+  if (!playback) {
+    if (curPath.includes("/login")) {
+      presenceData.details = "Đang đăng nhập";
+    } else if (search.startsWith("?s=")) {
+      presenceData.details = "Đang tìm kiếm";
+      presenceData.state = "Từ khoá: " + search.split("?s=")[1];
+    } else if (curPath.includes("/category")) {
+      const title = document.querySelector("title");
+      presenceData.details = "Đang tìm phim";
+      presenceData.state = title.textContent.split(" - ")[0];
+    } else if (curPath.includes("/hentai-uncensored")) {
+      presenceData.details = "Đang tìm phim Uncensored";
+    } else if (curPath.includes("/completed-hentai")) {
+      presenceData.details = "Đang tìm phim đã hoàn thành";
+    } else if (curPath.includes("/trailer-hentai")) {
+      presenceData.details = "Đang tìm trailer";
+    } else if (curPath.includes("/favorite-hentai")) {
+      presenceData.details = "Đang tìm danh đã thích";
+    } else if (curPath.includes("/images-gallery")) {
+      presenceData.details = "Đang tìm ảnh";
+    } else {
+      const title = document.querySelector(".anime-name>h1");
+      presenceData.details = title ? title.innerHTML : "Đang ở trang chủ";
+      if (title) presenceData.state = "Đang chọn tập";
+    }
+
+    if (!search.startsWith("?s=") && !curPath.includes("/category"))
+      delete presenceData.state;
+    delete presenceData.smallImageKey;
+
+    presenceData.startTimestamp = browsingStamp;
+    presence.setActivity(presenceData, false);
```

Comment:
shouldnt need to do startTimestamp again...

Reply:
It isn't necessary at all?

Reply:
You already do that on lines +31 to +35

Reply:
nope its not working when i remove that line
<img width="273" alt="image" src="https://user-images.githubusercontent.com/40353755/113184358-44efcc00-927f-11eb-9655-9f4eab275a98.png">

Reply:
hello can you approve?